adding key separator into spec to allow '.' in key values

henry-megarry opened the pull request.

JoshKCarroll commented:
Hey @henry-megarry, thanks for this PR! I can dig this change. I was wondering if you would be willing to make two changes:
* Add some additional test cases, both testing the utils for non-`.` keySeparator values as well as a couple of integration tests in `kazaam_int_test.go` with both explicitly setting `.` and setting non-standard keySeparator values.
* I'm wondering if we could do the setting of default keySeparator during initialization instead of hard coding it into all the util functions. Perhaps inspecting and updating the config in the [custom UnmarshalJSON() that we already have](https://github.com/qntfy/kazaam/blob/v3.4.7/spec.go#L21-L37)?

Appreciate it!!

henry-megarry commented:
cool I have added a few more tests.

I have also changed it to default it in the unmarshalJSON method. some of the tests use the getConfig() method in util_tests.go that I had to update to add "." as the key separator, just want to make sure thats alright?

JoshKCarroll commented on transform/util_test.go:
```diff
@@ -104,7 +104,35 @@ func TestSetJSONRaw(t *testing.T) {
+	}
+}
+
+func TestSetJSONRawNonDefaultKeySeporator(t *testing.T) {
```
Minor typo `Seporator` -> `Separator`

henry-megarry replied:
fixed

JoshKCarroll commented:
Yeah this looks great if you wanna just fix that typo ^ will approve and merge. Thanks!!

JoshKCarroll commented:
Tagged `v3.4.8`
